in_toto/verifylib.py

```diff
@@ -239,16 +239,17 @@ def verify_all_steps_signatures(layout, chain_link_dict):
 
   """
   for step in layout.steps:
-    keys_dict = {}
-
     # Find the according link for this step
     key_link_dict = chain_link_dict[step.name]
 
-    # Create the dictionary of keys for this step
-    for pubkeyid in step.pubkeys:
-      keys_dict[pubkeyid] = layout.keys[pubkeyid]
-
     for keyid, link in six.iteritems(key_link_dict):
+      keys_dict = {}
+
+      # Create the dictionary of keys for this step
+      for pubkeyid in step.pubkeys:
+          if pubkeyid == keyid:
+            keys_dict[pubkeyid] = layout.keys[pubkeyid]
```

Comment:
Isn't this check something you are [proposing in your PR](https://github.com/RohanBhirangi/in-toto/blob/225c0169f672fa16972ba893c67a39b149313343/in_toto/verifylib.py#L249-L250)? 

The only difference is that you would let a non-authorized key fail implicitly in the `Signable`'s `verify_signature` function due to an [empty signatures field](https://github.com/in-toto/in-toto/blob/develop/in_toto/models/common.py#L102) or an [empty key dictionary](https://github.com/in-toto/in-toto/blob/develop/in_toto/models/common.py#L108), and I suggest to let it fail explicitly. 

Maybe `SignatureVerificationError` is not the best exception type, maybe it should be something like `AuthorizationError`. 

In any case I'd say it does not hurt to re-check the authorization here, although it's already done in the `import_step_metadata_from_files_as_dict` function.